### PR TITLE
Stable Smoke Tests in Github Actions

### PR DESCRIPTION
**Pull request by whytheplatypus**
This PR stabilizes the smoke tests in our CI.

Before this the `volume` used by the postgres docker service could be manipulated by other steps or running jobs that caused smoke test failures. This removes the volume for CI runs.

Test Steps:
1. See that the Build Router check has passed, and the the Smoke Test step within that has passed.
2. Feel free to rerun the workflow to see it pass

Another method:
1. choose multiple passing commits from this branch
2. click rerun all jobs on for the Build Router workflow
3. see that they all pass and don't influence eachother
e.g.: 
- https://github.com/CDCgov/prime-reportstream/actions/runs/1472235578
- https://github.com/CDCgov/prime-reportstream/actions/runs/1467101999
- https://github.com/CDCgov/prime-reportstream/actions/runs/1463281590

Any other suggestions to assert that it won't cause the same trouble as before are greatly appreciated.

## Changes
- Use java setup action with caching
- Use gradle wrapper check to assert it's a valid jar
- Explicitly set up a local test environment for the CI run
- Add a postgres docker-compose with no persistent volume (I think this is what caused the earlier behavior)

## Fixes
- #2581 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- https://github.com/CDCgov/prime-reportstream/pull/3098
- https://github.com/CDCgov/prime-reportstream/issues/3104

### CODE REVIEW

**Review by MauriceReeves-usds — APPROVED** [review]

File: .github/workflows/build_hub.yml [hunk 1]
```diff
@@ -78,40 +77,43 @@ jobs:
...
-      - name: Dump docker logs on failure
-        if: failure()
-        uses: jwalton/gh-docker-logs@v2.0.2
-        with:
-          images: 'prime-router_settings,prime-router_prime_dev'
+      - name: Load settings
+        run: docker-compose -f docker-compose.yml up settings
 
+      - name: Correct sftp perms
+        run: docker-compose exec -T sftp chmod 777 /home/foo/upload
```
Comment (carlosfelix2) [inline .github/workflows/build_hub.yml:100]: Interesting.  You had to do this for Github?

File: prime-router/config/detekt/detekt.yml [hunk 2]
```diff
@@ -0,0 +1,796 @@
+build:
```
Comment (carlosfelix2) [inline prime-router/config/detekt/detekt.yml:1]: You are mixing work here.  This issue is for the smoke tests.  You should move this detekt work to another issue, so we can play with it and figure out if we want it.

File: prime-router/docker-compose.yml [hunk 3]
```diff
@@ -140,20 +140,17 @@ services:
       - prime-router_build
 
   web_receiver:
-    image: nginx:latest
-    ports:
-      - "8090:80"
+    image: node:14-alpine
```
Comment (carlosfelix2) [inline prime-router/docker-compose.yml:143]: What are this changes here?

File: prime-router/docker-compose.postgres.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Build environment
```
Comment (carlosfelix2) [inline prime-router/docker-compose.postgres.yml:1]: What is this?  How is this different from docker-compose.build.yml and why have the same config?

**Review by carlosfelix2 — CHANGES_REQUESTED** [review]

File: .github/workflows/build_hub.yml (hunk 1 above)
Comment (whytheplatypus) [reply .github/workflows/build_hub.yml:100]: yeah, the perms aren't translating when being mounted in the docker container.

File: prime-router/docker-compose.yml (hunk 3 above)
Comment (whytheplatypus) [reply prime-router/docker-compose.yml:143]: to make the frontend-react changes work. This was breaking with the removal of the `frontend` folder

File: prime-router/docker-compose.postgres.yml (hunk 4 above)
Comment (whytheplatypus) [reply prime-router/docker-compose.postgres.yml:1]: this doesn't use a persistent volume, which I'm theorizing was at least a part of the problem.

File: prime-router/docker-compose.yml (hunk 3 above)
Comment (JosiahSiegel) [reply prime-router/docker-compose.yml:143]: @whytheplatypus  you beat me to it :) [my-commit](https://github.com/CDCgov/prime-reportstream/pull/3062/commits/77f9a61a0e1eaeb3dd543d74ca82898fcafa26ad)

Will your changes fix the issue with running .cleanslate.sh? If so, I can delete my PR.

Comment (whytheplatypus) [reply prime-router/docker-compose.yml:143]: @JosiahSiegel oh awesome! lets use yours so it's not tied to the rest of this CI work.

Comment (JosiahSiegel) [reply prime-router/docker-compose.yml:143]: Sure. Once your commit is merged into master, i'll fetch your changes and test on my branch. Up to you if you want to back out any commits.

**Review by cwinters-usds — APPROVED** [review]
LGTM - I like standardizing on `gradlew` since it seems that's what everyone is using; thanks much for tracking this down and experimenting!

**Review by carlosfelix2 — APPROVED** [review]

**Pull request merged**